### ISSUE TITLE
Composer\InstalledVersions not found

### ISSUE THREAD

**Issue by martinssipenko** [humbug/php-scoper#532]
I'm trying to use scoper on my project where I use `Composer\InstalledVersions` class. After running scoper the class gets prefixed with `_PhpScoper2d7049acc8b0` and becomes `_PhpScoper2d7049acc8b0\Composer\InstalledVersions`. After running scoper I run `composer dump --classmap-authoritative` from within `build/` directory.

Now when I try to run my application I get an error:
`Uncaught Error: Class '_PhpScoperb8eeadf8a1cd\Composer\InstalledVersions'`

I've found that reference to this class is in `vendor/composer/autoload_classmap.php` file, but it's referencing a non-prefixed version:

```php
'Composer\\InstalledVersions' => $vendorDir . '/composer/InstalledVersions.php',
```

Is there anything I'm doing wrong?

**Comment by theofidry** [comment]
Hi, I'm not certain:
- which version of PHP-Scoper is it?
- what OS is it?
- what PHP version is it?
- are you loading the `vendor/scoper-autoload.php` file instead of the composer one?

**Comment by martinssipenko** [comment]
@theofidry 

- PHP Scoper version 0.14.1@023b231
- MacOS
- PHP 7.4.16
- I'm loading `vendor/autoload.php` in my application
- I'm using Composer version 2.1.5

**Comment by theofidry** [comment]
>I'm loading vendor/autoload.php in my application

This is the issue: for the scoped code you need to autoload the PHPScoper autoloader. It doesnt look clear from the doc in the "build without Box" section so it might be worth adding a note about it

**Comment by martinssipenko** [comment]
Just to be clear, I've actually tried using Box and encountered the same issue. I tried to pinpoint the issue by using Scoper without Box, which produced the same results.

**Comment by theofidry** [comment]
that is weird, but to keep in mind Box does more stuff so instead of trying to replicate what it does i would suggest to break down the issue as follows:

- see if it's a box issue or a PHP-Scoper issue: is the built PHAR working when building with box without PHP-Scoper?
- is the scoped app working fine outside the PHAR? (you can try `--debug` when building with Box it will dump the PHAR content in a temporary directory so you can execute the app outside the PHAR)

**Comment by martinssipenko** [comment]
I've narrowed it down a bit, turns out that when I run the application that was produced using `--debug` option, one that's located in `.box_dump/` directory, the composer uses class map from `autoload_static.php` file, but similar to what I said before the value there is without the prefix:
```php
'Composer\\InstalledVersions' => __DIR__ . '/..' . '/composer/InstalledVersions.php',
```

If I use Box without scoper it works as expected.

**Comment by theofidry** [comment]
do you have a reproducer? I think there is more to it because both Box and PHP-Scoper build themselves in a similar fashion

**Comment by martinssipenko** [comment]
Wdym? A repo where this can be observed?

**Comment by theofidry** [comment]
Yes, something I could easily check out locally ideally

**Comment by martinssipenko** [comment]
here is a reproducer https://github.com/martinssipenko/php-scoper-issue
clone it, composer install and try running box

**Comment by theofidry** [comment]
thanks! will try to check it out during the week

**Comment by weirdan** [comment]
This now affects Psalm as well: vimeo/psalm#7329. @theofidry, did you have any luck looking into this? Even though we're using an outdated box version in Psalm, upgrading didn't change a thing as far as this issue is concerned.

**Comment by theofidry** [comment]
This should be fixed with 0.16. If that's not the case please open a new issue
